src/AutoRest.CSharp/Mgmt/Generation/MgmtClientBaseWriter.cs

```diff
@@ -254,11 +254,55 @@ protected virtual void WriteChildResourceEntries()
                     {
                         WriteResourceCollectionEntry(resource.ResourceCollection, signature);
                     }
+
+                    if (!(This is MgmtExtensionClient)) // we don't need to generate `Get{Resource}` methods in ExtensionClient
```

Comment:
Do we want to add `Get{Resource}` to Extensions? If no, the codes can be further simplified.